src/App.tsx

```diff
@@ -1,56 +1,80 @@
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+
+/* eslint-disable-next-line */
+function debonce(callback: Function, delay: number) {
+  let timerId = 0;
+
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setInterval(() => {
+      callback(...args);
+    }, delay);
+  };
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const { name, born, died } = selectedPerson || {};
+
+  const applyQuery = useCallback(debonce(setAppliedQuery, 1000), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const people = useMemo((): Person[] => {
+    const normalizedQuery = appliedQuery.trim().toLowerCase();
+
+    return peopleFromServer
+      .filter(person => person.name.toLowerCase().includes(normalizedQuery));
+  }, [peopleFromServer, appliedQuery]);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson ? (
+          `${name} (${born} - ${died})`
+        ) : (
+          'No selected person'
+        )}
       </h1>
 
-      <div className="dropdown is-active">
+      <div className={classNames('dropdown', {
+        'is-active': !!appliedQuery,
+      })}
+      >
         <div className="dropdown-trigger">
           <input
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={handleQueryChange}
           />
         </div>
 
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
+        {appliedQuery && !people.length ? (
```

Comment:
Than I select person the dropdown is open for another second
![image](https://github.com/mate-academy/react_autocomplete/assets/54936044/f9dcdfd5-45c0-4eaa-a3c8-1cf6ab00593a)